core/chains/cosmos/chain_set.go

```diff
@@ -0,0 +1,116 @@
+package cosmos
+
+import (
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+
+	"github.com/smartcontractkit/sqlx"
+
+	"github.com/smartcontractkit/chainlink-cosmos/pkg/cosmos/adapters"
+	coscfg "github.com/smartcontractkit/chainlink-cosmos/pkg/cosmos/config"
+	"github.com/smartcontractkit/chainlink-cosmos/pkg/cosmos/db"
+
+	"github.com/smartcontractkit/chainlink/core/chains"
+	"github.com/smartcontractkit/chainlink/core/chains/cosmos/types"
+	coreconfig "github.com/smartcontractkit/chainlink/core/config"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/keystore"
+	"github.com/smartcontractkit/chainlink/core/services/pg"
+)
+
+var (
+	// ErrChainIDEmpty is returned when chain is required but was empty.
+	ErrChainIDEmpty = errors.New("chain id empty")
+	// ErrChainIDInvalid is returned when a chain id does not match any configured chains.
+	ErrChainIDInvalid = errors.New("chain id does not match any local chains")
+)
+
+// ChainSetOpts holds options for configuring a ChainSet.
+type ChainSetOpts struct {
+	Config           coreconfig.BasicConfig
+	Logger           logger.Logger
+	DB               *sqlx.DB
+	KeyStore         keystore.Cosmos
+	EventBroadcaster pg.EventBroadcaster
+	ORM              types.ORM
+}
+
+func (o *ChainSetOpts) Validate() (err error) {
+	required := func(s string) error {
+		return errors.Errorf("%s is required", s)
+	}
+	if o.Config == nil {
+		err = multierr.Append(err, required("Config"))
+	}
+	if o.Logger == nil {
+		err = multierr.Append(err, required("Logger'"))
+	}
+	if o.DB == nil {
+		err = multierr.Append(err, required("DB"))
+	}
+	if o.KeyStore == nil {
+		err = multierr.Append(err, required("KeyStore"))
+	}
+	if o.EventBroadcaster == nil {
+		err = multierr.Append(err, required("EventBroadcaster"))
+	}
+	if o.ORM == nil {
+		err = multierr.Append(err, required("ORM"))
+	}
+	return
+}
+
+func (o *ChainSetOpts) ORMAndLogger() (chains.ORM[string, *db.ChainCfg, db.Node], logger.Logger) {
+	return o.ORM, o.Logger
+}
+
+// https://app.shortcut.com/chainlinklabs/story/33622/remove-legacy-config
```

Comment:
I think most of us no longer have access to shortcut so new PRs should point to JIRA. Is there a JIRA ticket that was ported over?

Reply:
Yes, but it is already completed. This should be removed.
https://smartcontract-it.atlassian.net/browse/BCF-1401